servicetalk-http-router-predicate/src/main/java/io/servicetalk/http/router/predicate/InOrderRouter.java

```diff
@@ -68,9 +70,9 @@
                 StreamingHttpService service = pair.service();
                 final HttpExecutionStrategy strategy = pair.routeStrategy();
                 if (strategy != null) {
-                    Executor se = strategy.executor();
-                    Executor executor = null == se ? ctx.executionContext().executor() : se;
-                    service = strategy.offloadService(executor, service);
+                    Executor executor = ctx.executionContext().executor();
+                    service = StreamingHttpServiceToOffloadedStreamingHttpService.offloadService(strategy,
+                            executor, () -> Thread.currentThread() instanceof IoThreadFactory.IoThread, service);
```

Comment:
can we leverage `BooleanSupplier shouldOffload(IoExecutor executor)` here?

Reply:
yes